ISLANDORA-2053: Update travis.yml to force PHP 5.3 to run under Ubuntu Precise

DiegoPino opened the pull request:
**ISLANDORA-2053**: (https://jira.duraspace.org/browse/ISLANDORA-2053)
http://irclogs.islandora.ca/2017-09-05.html
https://github.com/travis-ci/travis-ci/issues/2963
https://github.com/Islandora/islandora/pull/683

As of September 2017, Travis-CI has removed support for PHP 5.3 (and 5.2) for their Ubuntu Trusty Images, forcing us, in order  to keep Drupal's compatibility matrix tested, to run Travis-CI test for 5.3 on Ubuntu Precise


# What does this Pull Request do?

This pull Fixes our failing Travis-CI tests: 
Should allow PHP 5.3.3 to run under Ubuntu Precise and Higher versions under Ubuntu Trusty.
Changes the way we define which PHP versions and Fedora ones run in Travis. Also, brings this module up to date testing PHP 5.6! 7.0 and 7.1 plus the new "islandora_crossref_citations" submodule.

# What's new?
We Moved PHP versions and Fedora ones into matrix section of the Travis YAML file.  We added PHP versions and a new submodule.

# How should this be tested?
Travis-CI should pass and everything single PHP/FEDORA version combination should look good and very green.

# Interested parties
@Islandora/7-x-1-x-committers @bondjimbond

DiegoPino committed:
force PHP 5.3 to run under Ubuntu Precise

And bring tests up to date (islandora_crossref_citations) +PHP 5.6 , 7.0 and 7.1

DiegoPino commented:
@bondjimbond do you have some time today?
After enabling actual dcs check on the new submodule you contributed a few days ago there is an error
https://travis-ci.org/Islandora/islandora_badges/jobs/272963908#L1007
Could you please make a quick fix on that? So we can move on this? Thanks

DiegoPino commented:
@bondjimbond ping! Can you please fix your DCS from that previous pull?

DiegoPino committed:
add notifications

whikloj commented:
There is still a problem in the code, but also those versions showing green seem to be missing PHP Codesniffer.
https://travis-ci.org/Islandora/islandora_badges/jobs/275570844#L724-L737

DiegoPino committed:
DSC. I need travis to pass, sorry

had to fix parent (recent) pull, but close to code freeze and need this
one passing.
